tket/src/Transformations/CliffordReductionPass.cpp

```diff
@@ -410,8 +410,12 @@ Subcircuit CliffordReductionPass::substitute(
   // edges.
   EdgeList future_edges;
   VertexSet v_frontier;
+  // keep track of visited vertices
+  VertexSet visited;
   for (unsigned qi = 0; qi < q_width; ++qi) {
-    v_frontier.insert(circ.target(out_edges[qi]));
+    Vertex target = circ.target(out_edges[qi]);
+    visited.insert(target);
+    v_frontier.insert(target);
   }
```

Comment:
Maybe this would be easier to read if we just assigned
```c++
VertextSet visited = v_frontier;
```
after the original loop? Up to you.

Reply:
ahh, sorry I didn't see this...